common/src/main/java/org/apache/comet/parquet/FileReader.java

```diff
@@ -128,6 +135,48 @@ public FileReader(InputFile file, ParquetReadOptions options, ReadOptions cometO
     this(file, null, options, cometOptions, null);
   }
 
+  /** This constructor is called from Apache Iceberg. */
+  public FileReader(
+      Path path,
+      Configuration conf,
+      ReadOptions cometOptions,
+      Map<String, String> properties,
+      Long start,
+      Long length,
+      byte[] fileEncryptionKey,
+      byte[] fileAADPrefix)
+      throws IOException {
+    ParquetReadOptions options =
+        buildParquetReadOptions(conf, properties, start, length, fileEncryptionKey, fileAADPrefix);
+    this.converter = new ParquetMetadataConverter(options);
+    this.file = CometInputFile.fromPath(path, conf);
+    this.f = file.newStream();
+    this.options = options;
+    this.cometOptions = cometOptions;
+    this.metrics = null;
+    try {
+      this.footer = readFooter(file, options, f, converter);
```

Comment:
The footer should be available already in Iceberg? Can we avoid having to read it twice?

Reply:
Yes, the footer is already available from iceberg, but I can't pass `ParquetMetadata` from Iceberg to Comet. Maybe I can convert `ParquetMetadata` to JSON string and pass it to Comet, and then construct a `ParquetMetadata` from JSON string.

Reply:
I initially thought I can do something like this:
```
       ParquetMetadata footer = reader.getFooter();
       String footerInJson = ParquetMetadata.toJSON(footer);
       ParquetMetadata footer2 = ParquetMetadata.fromJSON(footerInJson);
```
but this doesn't work. I got 
```
java.lang.RuntimeException: shaded.parquet.com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.apache.parquet.hadoop.metadata.ParquetMetadata` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: (StringReader); line: 1, column: 2]
```
I will see if there are other ways to re-construct ParquetMetadata.